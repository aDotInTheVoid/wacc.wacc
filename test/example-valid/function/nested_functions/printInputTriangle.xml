<unit>
  <fn>
    <name>f</name>
    <args><arg>0</arg></args>
    <body>
      <if>
        <cond><local>0</local><num>0</num><op>eq</op></cond>
        <when/>
        <else>
          <addr_local>1</addr_local>
          <local>0</local>
          <assign_do/>
          <while>
            <cond><local>1</local><num>0</num><op>gt</op></cond>
            <body>
              <str>"-"</str>
              <print><kind>str</kind><newline>false</newline></print>
              <addr_local>1</addr_local>
              <local>1</local>
              <num>1</num>
              <op>sub</op>
              <assign_do/>
            </body>
          </while>
          <str>""</str>
          <print><kind>str</kind><newline>true</newline></print>
          <addr_local>2</addr_local>
          <local>0</local>
          <num>1</num>
          <op>sub</op>
          <call_func><name>f</name><nargs>1</nargs></call_func>
          <assign_do/>
        </else>
      </if>
      <num>0</num>
      <return/>
    </body>
  </fn>
  <main>
    <str>"Please enter the size of the triangle to print: "</str>
    <print><kind>str</kind><newline>true</newline></print>
    <addr_local>0</addr_local>
    <num>0</num>
    <assign_do/>
    <addr_local>0</addr_local>
    <read><ty>int</ty></read>
    <addr_local>1</addr_local>
    <local>0</local>
    <call_func><name>f</name><nargs>1</nargs></call_func>
    <assign_do/>
  </main>
</unit>