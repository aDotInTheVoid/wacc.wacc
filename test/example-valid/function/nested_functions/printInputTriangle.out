Please enter the size of the triangle to print:
#input#
#output#
